### PR TITLE
Delete Recording

### PR DESCRIPTION
**Pull request by hadicheaito1**
## Description

- Allows you to delete a recording that is on the 'Recordings' page
- Click on fa icon to delete

## Testing Steps

1. Pull code
2. Create room
3. Create recording for that room
4. Go to 'Recordings' page
5. Click on delete icon for that recording
6. Check in db using rails console to ensure that recording is deleted

## Screenshots (if appropriate):

image
image

### CODE REVIEW

File: app/controllers/api/v1/recordings_controller.rb [hunk 1]
```diff
@@ -14,6 +14,12 @@ def index
         render_json(data: recordings, status: :ok, include: :formats)
       end
 
+      def destroy
+        Recording.destroy(params[:id])
+
```
Comment (farhatahmad) [inline app/controllers/api/v1/recordings_controller.rb:19]: You need to actually delete the recording here

File: app/javascript/components/recordings/Recordings.jsx [hunk 2]
```diff
@@ -44,6 +46,9 @@ export default function Recordings() {
                       <div key={format.id}> {format.recording_type} </div>
                     ))}
                   </td>
+                  <td>
+                    <FontAwesomeIcon className="" icon={faTrashCan} size="lg" onClick={() => handleDeleteRecording({ recordId: recording.id })} />
```
Comment (farhatahmad) [inline app/javascript/components/recordings/Recordings.jsx:50]: I think this should pop a modal up to be safe

Comment (scouillard) [reply app/javascript/components/recordings/Recordings.jsx:50]: Heads up we have a Modal template in components/shared.

I did a "font-awesome" button variant as well if you can check it out in #3447 application.scss if you want (not merged yet).

image

File: app/controllers/api/v1/recordings_controller.rb (hunk 1 above)
Comment (hadicheaito1) [reply app/controllers/api/v1/recordings_controller.rb:19]: Wouldn't we need it to be destroy in order to remove the formats that belong it as well? Or is there another reason?

Comment (farhatahmad) [reply app/controllers/api/v1/recordings_controller.rb:19]: As in, you need to make a request to BigBlueButton to destroy the recording

File: app/javascript/components/recordings/Recordings.jsx (hunk 2 above)
Comment (hadicheaito1) [reply app/javascript/components/recordings/Recordings.jsx]: Thanks for the tip Samuel helped a lot.

File: app/controllers/api/v1/recordings_controller.rb (hunk 1 above)
Comment (hadicheaito1) [reply app/controllers/api/v1/recordings_controller.rb:19]: Will do this in next PR

**Review by farhatahmad — APPROVED** [review]

**Pull request merged**